include/albatross/src/stats/incomplete_gamma.hpp

```diff
@@ -67,58 +67,46 @@ inline double incomplete_gamma_quadrature_recursive(double lb, double ub,
   }
 }
 
-inline double incomplete_gamma_quadrature_lb(double a, double z) {
-  if (a > 1000) {
-    return std::max(0., std::min(z, a) - 11 * sqrt(a));
-  } else if (a > 800) {
-    return std::max(0., std::min(z, a) - 11 * sqrt(a));
-  } else if (a > 500) {
-    return std::max(0., std::min(z, a) - 10 * sqrt(a));
-  } else if (a > 300) {
-    return std::max(0., std::min(z, a) - 10 * sqrt(a));
-  } else if (a > 100) {
-    return std::max(0., std::min(z, a) - 9 * sqrt(a));
-  } else if (a > 90) {
-    return std::max(0., std::min(z, a) - 9 * sqrt(a));
-  } else if (a > 70) {
-    return std::max(0., std::min(z, a) - 8 * sqrt(a));
-  } else if (a > 50) {
-    return std::max(0., std::min(z, a) - 7 * sqrt(a));
-  } else if (a > 40) {
-    return std::max(0., std::min(z, a) - 6 * sqrt(a));
-  } else if (a > 30) {
-    return std::max(0., std::min(z, a) - 5 * sqrt(a));
-  }
-  return std::max(0., std::min(z, a) - 4 * sqrt(a));
-}
+inline std::pair<double, double> incomplete_gamma_quadrature_bounds(double a,
+                                                                    double z) {
 
-inline double incomplete_gamma_quadrature_ub(double a, double z) {
-  if (a > 1000) {
-    return std::min(z, a + 10 * sqrt(a));
-  } else if (a > 800) {
-    return std::min(z, a + 10 * sqrt(a));
+  if (a > 800) {
+    return std::make_pair(std::max(0., std::min(z, a) - 11 * sqrt(a)),
+                          std::min(z, a + 10 * sqrt(a)));
   } else if (a > 500) {
-    return std::min(z, a + 9 * sqrt(a));
+    return std::make_pair(std::max(0., std::min(z, a) - 10 * sqrt(a)),
+                          std::min(z, a + 9 * sqrt(a)));
```

Comment:
Is this necessary if the next conditional will cover this case with the same result?
Is this the desired behavior or is there a typo in the bounds.

Reply:
Nah I just missed it.  all this is taken from https://github.com/kthohr/gcem/blob/master/include/gcem_incl/incomplete_gamma.hpp#L76-L113